vcpkg/users/assetcaching.md

```diff
@@ -57,3 +57,70 @@ See also the [binary caching documentation for Azure Blob Storage](binarycaching
 Syntax: `x-block-origin`
 
 Disables use of the original URLs in case the mirror does not have the file available.
+
+#### `x-script`
+
+Dispatches to an external tool to fetch the asset. Within the template, `{url}` will be replaced by the original url, `{sha512}` will be replaced by the SHA512 value, and `{dst}` will be replaced by the output path to save to. These substitutions will all be properly shell escaped, so an example template would be: `curl -L {url} --output {dst}`. `{{` will be replaced by `}` and `}}` will be replaced by `}` to avoid expansion. Note that this will be executed inside the build environment, so the PATH and other environment variables will be modified by the triplet.
+
+## Example
```

Comment:
Markdown headers (lines beginning with `#`) are hierarchical -- this means that this example section needs to be
```suggestion
##### Example
```
So it's "under" the `x-script` section.